internal/gitserver/stream_hunks.go

```
@@ -0,0 +1,233 @@
+	for {
+		if err := ctx.Err(); err != nil {
+			close(p.done)
+			break
+		}
+
+		// Do we have more to read?
+		if !p.sc.Scan() {
+			if cur != nil {
+				if h, ok := p.commits[string(cur.CommitID)]; ok {
+					cur.CommitID = h.CommitID
+					cur.Author = h.Author
+					cur.Message = h.Message
+				}
+				// If we have an ongoing entry, send it.
+				p.hunksCh <- hunkResult{hunk: cur}
+			}
+			break
+		}
+
+		var done bool
+		var err error
+		// Read line from git blame, in porcelain format
+		annotation, fields := p.scanLine()
+
+		// On the first read, we have no hunk and the first thing we read is an entry.
+		if cur == nil {
+			cur, err = parseEntry(annotation, fields)
+			if err != nil {
+				p.hunksCh <- hunkResult{err: err}
+			}
+			continue
+		}
+
+		// After that, we're either reading extras, or a new entry.
+		done, err = parseExtra(cur, annotation, fields)
+		if err != nil {
+			p.hunksCh <- hunkResult{err: err}
+		}
+		// If we've finished reading extras, we're looking at a new entry.
+		if done {
+			if h, ok := p.commits[string(cur.CommitID)]; ok {
+				cur.CommitID = h.CommitID
+				cur.Author = h.Author
+				cur.Message = h.Message
+			} else {
+				p.commits[string(cur.CommitID)] = cur
+			}
+
+			p.hunksCh <- hunkResult{hunk: cur}
+
+			cur, err = parseEntry(annotation, fields)
+			if err != nil {
+				p.hunksCh <- hunkResult{err: err}
+			}
+		}
+	}
+
+	// If there is an error from the scanner, send it back.
+	if err := p.sc.Err(); err != nil {
+		p.hunksCh <- hunkResult{err: err}
+	}
+	close(p.done)
+}
+
+// parseEntry turns a `67b7b725a7ff913da520b997d71c840230351e30 10 20 1` line from
+// git blame into a hunk.
+func parseEntry(rev string, content string) (*Hunk, error) {
+	fields := strings.Split(content, " ")
+	if len(fields) != 3 {
+		return nil, errors.Errorf("HERE Expected at least 4 parts to hunkHeader, but got: '%s %s'", rev, content)
+	}
+
+	resultLine, _ := strconv.Atoi(fields[1])
+	numLines, _ := strconv.Atoi(fields[2])
```

Should we check these errors? Seems like swallowing these could cause some confusing behavior in the future